### ISSUE TITLE
Scopes: Adds scope to control if a user edit a declaration of another user

### ISSUE THREAD

**Issue by jpye-finch** [opencrvs/opencrvs-core#7422]
### Description
In some countries only the original "preparer" of the declaration form can update it once it is "In Review" or "Requires Updates" 

This ticket is to:
- Add scope record.declaration-edit
- If a user has this scope then the 'Change' links show for the user
- This lets the edit the declaration created by another user
- A user can edit their declaration by default (this scope is not required)

### ACs
GIVEN i have the record.declaration-edit scope
WHEN reviewing a declaration with the status "Incomplete, "Validated", "In Review" or "Requires Updates" 
THEN I can edit the declaration (Change links show)

**! Don't need record.declaration-edit scope to edit my own records**
GIVEN i **don't** have the record.declaration-edit scope
WHEN reviewing a declaration **started by me** 
THEN I can edit my declaration (Change links show)

**Comment by SyedaAfrida** [comment]
@jpye-finch @Alta-Nel edit can also be done reviewing a declaration with the status "Validated" and "Incomplete". Isn't it?

**Comment by SyedaAfrida** [comment]
@jpye-finch can you also check your 2nd AC, seems confusing

**Comment by jpye-finch** [comment]
Yes apoligies this is about not needing record.declaration-edit scope to edit my own records
